### ISSUE TITLE
OverrideUserAgent and AppendUserAgent not working for Android

### ISSUE THREAD

**Issue by ajay1271** [apache/cordova-plugin-inappbrowser#759]
# Bug Report

## Problem

Tried changing the webview UserAgent in config.xml.  used both

```
<preference name="OverrideUserAgent" value="Mozilla/5.0 Google" />
<preference name="AppendUserAgent" value="Mozilla/5.0 Google" /> 
```

in config.xml for both android and iOS preferences. Its working fine for iOS but not working for android. 

### What is expected to happen?
 it should should change the userAgent to given OverrideUserAgent value

### What does actually happen?
Its not changing for android. 

## Information

Tried debugging the generated code. Its giving null values for both OverrideUserAgent and AppendUserAgent preferences in SystemWebViewEngine.Java class

### Command or Code

```
**config.xml**

<?xml version='1.0' encoding='utf-8'?>
<widget id="xxxxxx" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0"
android-versionCode="xxxxxx" version="xxxxxx">
  <name>xxxxxx</name>
  <description>
    xxxxxx
  </description>
  <author email="xxxxxx" href="xxxxxx">
  xxxxxx
  </author>
  <content src="index.html" />
  <access origin="*" />
  <allow-navigation href="*" />
  <allow-intent href="http://*/*" />
  <allow-intent href="https://*/*" />
  <allow-intent href="tel:*" />
  <allow-intent href="sms:*" />
  <allow-intent href="mailto:*" />
  <allow-intent href="geo:*" />
  <platform name="android">
    <allow-intent href="market:*" />   
    <preference name="OverrideUserAgent" value="Mozilla/5.0 Google" />
    <preference name="android-targetSdkVersion" value="29" />
  </platform>
  <platform name="ios">
    <allow-intent href="itms:*" />
    <allow-intent href="itms-apps:*" />
    <preference name="orientation" value="portrait" />
    <preference name="StatusBarOverlaysWebView" value="false" />
    <preference name="StatusBarBackgroundColor" value="#cc0033" />
    <preference name="WKWebViewOnly" value="true" />
    <feature name="CDVWKWebViewEngine">
        <param name="ios-package" value="CDVWKWebViewEngine" />
    </feature>
    <preference name="CordovaWebViewEngine" value="CDVWKWebViewEngine" />
    <preference name="OverrideUserAgent" value="Mozilla/5.0 Google" />
  </platform>
  
  <plugin name="cordova-plugin-keyboard" spec="^1.1.5" />
  <plugin name="cordova-plugin-statusbar" spec="^2.3.0" />
  <plugin name="cordova-plugin-whitelist" spec="1" />
  <plugin name="cordova-plugin-customurlscheme" spec="^4.3.0">
    <variable name="URL_SCHEME" value="xxxxios" />
  </plugin>
  <plugin name="cordova-plugin-fastclick" spec="https://github.com/chemerisuk/cordova-plugin-fastclick.git" />
  <engine name="ios" spec="~4.5.4" />
</widget>
```

### Environment, Platform, Device

Android

### Version information

Android targeted SDK 29

cordova-lib@9.0.1
cordova-plugin-inappbrowser 4.0.0

**platforms list:**
android@8.1.0 
ios@4.5.5

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by breautek** [comment]
Based on https://github.com/apache/cordova-android/blob/master/framework/src/org/apache/cordova/ConfigXmlParser.java

I don't think android supports having preferences inside the `<platform>` tags... Which I think is problematic. But does it work if you move the preference outside of the `<platform>` tags?

**Comment by ajay1271** [comment]
> Based on https://github.com/apache/cordova-android/blob/master/framework/src/org/apache/cordova/ConfigXmlParser.java
> 
> I don't think android supports having preferences inside the `<platform>` tags... Which I think is problematic. But does it work if you move the preference outside of the `<platform>` tags?

Hi, thanks for the reply. I tried moving the appendUserAgent preference outside the platform tag. Still the same issue

**Comment by breautek** [comment]
Both of these properties works for me on `cordova-android@9` inside and outside the `<platform>` tag, can you try upgrading the platform?

**Comment by shawnmbradley** [comment]
I too am having this same issue with "cordova-android": "^9.0.0".

Everything works as expected on iOS devices and on my Kindle Fire.  Although on a device where our MDM is force Chrome Engine for the webkit it is not working.  Could that be the issue?

**Comment by MateiNenciu** [comment]
Same issue with cordova-android 9.0. On iOS works perfectly, but on Android works the very first time I open the inAppBrowser. Afterwards I need to reinstall the app to work again. I too have the AppendUserAgent property outside platform tags.

**Comment by laurentperez** [comment]
Same bug. Not working.

Either Emulator with "Dalvik/2.1.0 (Linux; U; Android 10; Android SDK built for x86 Build/QSR1.190920.001)" or real device with  "Dalvik/2.1.0 (Linux; U; Android 8.0.0; PRA-LX1 Build/HUAWEIPRA-LX1)"

**Comment by harlamoff** [comment]
Same bug. 
ios - work
android - bug

Cordova Android 11.0.0

**Comment by diogenesricardo** [comment]
We still have this bug in 2025?

**Comment by diogenesricardo** [comment]
> We still have this bug in 2025?

I found the problem. It seems to be a conflict with the cordova-plugin-advanced-http plugin that prevents the config.xml settings for AppendUserAgent and OverrideUserAgent from being honored.
I still haven't found a solution.

@NiklasMerz

**Comment by diogenesricardo** [comment]
> 
> Naturally the preferences needs to get mapped to settings and based on my 2020 comment, that does occur, at least in whatever test app I decided to build at that time. If you're using a plugin that does http networking, it will be up to that plugin to read the preferences and honour setting the user agent header appropriately.
> 
> But that doesn't seem related to this issue because the in app browser content shouldn't be using plugins (or even have access to plugin APIs). Unless if the plugin itself is overwriting the `AppendUserAgent` and/or `OverrideUserAgent` preferences.

Exactly, that's why I believe this issue shouldn't even be open. It's not a problem with cordova-plugin-inappbrowser
